File: app/src/main/java/jogja/telkom/csrassistant/Adapter/RecAdapter.java
package jogja.telkom.csrassistant.Adapter;

import android.content.Context;
import android.view.Gravity;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.TextView;
import android.widget.Toast;

import java.util.ArrayList;

import androidx.core.content.ContextCompat;
import androidx.recyclerview.widget.RecyclerView;
import jogja.telkom.csrassistant.R;

public class RecAdapter extends RecyclerView.Adapter<RecAdapter.MyViewHolder> {
    private ArrayList<Instruksi> instruksis;

    public RecAdapter(ArrayList<Instruksi> instruksis) {
        this.instruksis = instruksis;
    }

    @Override
    public RecAdapter.MyViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View itemView = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_recycler, parent, false);
        return new MyViewHolder(itemView);
    }

    @Override
    public void onBindViewHolder(RecAdapter.MyViewHolder holder, int position) {
        Instruksi instruksi= instruksis.get(position);
        holder.textView_instruksi.setText(instruksi.getInstruksi());
        //
        int noOfChildTextViews = holder.linearLayout_isiInstruksi.getChildCount();
        int noOfChild = instruksi.getChildDataItems().size();
        if (noOfChild < noOfChildTextViews) {
            for (int index = noOfChild; index < noOfChildTextViews; index++) {
                TextView currentTextView = (TextView) holder.linearLayout_isiInstruksi.getChildAt(index);
                currentTextView.setVisibility(View.GONE);
            }
        }
        for (int textViewIndex = 0; textViewIndex < noOfChild; textViewIndex++) {
            TextView currentTextView = (TextView) holder.linearLayout_isiInstruksi.getChildAt(textViewIndex);
            currentTextView.setText(instruksi.getChildDataItems().get(textViewIndex).getIsiInstruksi());
        }
    }

    @Override
    public int getItemCount() {
        return instruksis.size();
    }

    class MyViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
        private Context context;
        private TextView textView_instruksi;
        private LinearLayout linearLayout_isiInstruksi;

        MyViewHolder(View itemView) {
            super(itemView);
            context = itemView.getContext();
            textView_instruksi = itemView.findViewById(R.id.item_instruksi);
            linearLayout_isiInstruksi = itemView.findViewById(R.id.sub_item_isi);
            linearLayout_isiInstruksi.setVisibility(View.GONE);
            int intMaxNoOfChild = 0;
            for (int index = 0; index < instruksis.size(); index++) {
                int intMaxSizeTemp = instruksis.get(index).getChildDataItems().size();
                if (intMaxSizeTemp > intMaxNoOfChild) intMaxNoOfChild = intMaxSizeTemp;
            }
            //SET TEXT FORMAT UNTUK ISI
            for (int indexView = 0; indexView < intMaxNoOfChild; indexView++) {
                TextView textView = new TextView(context);
                textView.setId(indexView);
                textView.setPadding(16, 20, 0, 20);
                textView.setGravity(Gravity.LEFT);
                textView.setBackground(ContextCompat.getDrawable(context, R.drawable.background_sub_module_text));
                LinearLayout.LayoutParams layoutParams = new LinearLayout.LayoutParams(LinearLayout.LayoutParams.MATCH_PARENT, LinearLayout.LayoutParams.WRAP_CONTENT);
                textView.setOnClickListener(this);
                linearLayout_isiInstruksi.addView(textView, layoutParams);
            }
            textView_instruksi.setOnClickListener(this);
        }

        @Override
        public void onClick(View view) {
            if (view.getId() == R.id.item_instruksi) {
                if (linearLayout_isiInstruksi.getVisibility() == View.VISIBLE) {
                    linearLayout_isiInstruksi.setVisibility(View.GONE);
                } else {
                    linearLayout_isiInstruksi.setVisibility(View.VISIBLE);
                }
            } else {
                TextView textViewClicked = (TextView) view;
                Toast.makeText(context, "" + textViewClicked.getText().toString(), Toast.LENGTH_SHORT).show();
            }
        }
    }
}

File: app/src/main/java/jogja/telkom/csrassistant/Adapter/RecAdapter.kt
package jogja.telkom.csrassistant.Adapter

import android.content.Context
import android.graphics.Color
import android.graphics.Typeface
import android.view.Gravity
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.LinearLayout
import android.widget.TextView
import android.widget.Toast
import androidx.core.content.ContextCompat
import androidx.recyclerview.widget.RecyclerView
import jogja.telkom.csrassistant.R
import java.util.*

class RecAdapter(private val instruksis: ArrayList<Instruksi>) : RecyclerView.Adapter<RecAdapter.MyViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecAdapter.MyViewHolder {
        val itemView = LayoutInflater.from(parent.context).inflate(R.layout.item_recycler, parent, false)
        return MyViewHolder(itemView)
    }

    override fun onBindViewHolder(holder: RecAdapter.MyViewHolder, position: Int) {
        val instruksi = instruksis[position]
        holder.textView_instruksi.text = instruksi.instruksi
        //
        val noOfChildTextViews = holder.linearLayout_isiInstruksi.childCount
        val noOfChild = instruksi.childDataItems!!.size
        if (noOfChild < noOfChildTextViews) {
            for (index in noOfChild until noOfChildTextViews) {
                val currentTextView = holder.linearLayout_isiInstruksi.getChildAt(index) as TextView
                currentTextView.visibility = View.GONE
            }
        }
        for (textViewIndex in 0 until noOfChild) {
            val currentTextView = holder.linearLayout_isiInstruksi.getChildAt(textViewIndex) as TextView
            currentTextView.text = instruksi.childDataItems!![textViewIndex].isiInstruksi
        }
    }

    override fun getItemCount(): Int {
        return instruksis.size
    }

    inner class MyViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView), View.OnClickListener {
        private val context: Context
        val textView_instruksi: TextView
        val linearLayout_isiInstruksi: LinearLayout

        init {
            context = itemView.context
            textView_instruksi = itemView.findViewById(R.id.item_instruksi)
            linearLayout_isiInstruksi = itemView.findViewById(R.id.sub_item_isi)
            linearLayout_isiInstruksi.visibility = View.GONE
            var intMaxNoOfChild = 0
            for (index in instruksis.indices) {
                val intMaxSizeTemp = instruksis[index].childDataItems!!.size
                if (intMaxSizeTemp > intMaxNoOfChild) intMaxNoOfChild = intMaxSizeTemp
            }

            //SET TEXT FORMAT UNTUK ISI
            for (indexView in 0 until intMaxNoOfChild) {
                val textView = TextView(context)
                textView.id = indexView
                textView.setPadding(16, 20, 16, 20)
                textView.gravity = Gravity.LEFT
                textView.textSize = 18F
                textView.setTextColor(Color.BLACK)
                textView.typeface = Typeface.createFromAsset(context.assets, "res/font/roboto_regular.ttf")
                textView.background = ContextCompat.getDrawable(context, R.drawable.background_sub_module_text)
                val layoutParams = LinearLayout.LayoutParams(LinearLayout.LayoutParams.MATCH_PARENT, LinearLayout.LayoutParams.WRAP_CONTENT)
                textView.setOnClickListener(this)
                linearLayout_isiInstruksi.addView(textView, layoutParams)
            }
            textView_instruksi.setOnClickListener(this)
        }

        override fun onClick(view: View) {
            if (view.id == R.id.item_instruksi) {
                if (linearLayout_isiInstruksi.visibility == View.VISIBLE) {
                    linearLayout_isiInstruksi.visibility = View.GONE
                } else {
                    linearLayout_isiInstruksi.visibility = View.VISIBLE
                }
            } else {
                val textViewClicked = view as TextView
                Toast.makeText(context, "" + textViewClicked.text.toString(), Toast.LENGTH_SHORT).show()
            }
        }
    }
}
